### PR TITLE
Use Maven Publish Plugin for the core module

### PR DESCRIPTION
**Pull request by Andati**
**Description:**
Add support to publish core.
You can publish core locally via command: `./gradlew :core:publishToMavenLocal` to include in your project for local testing

**Type**
Builds

### CODE REVIEW

**Comment by jingtang10** [comment]
Hi @Andati is this ready for review? thanks!

**Comment by Andati** [comment]
@jingtang10 I see that the master has changed, let me update the pr first

File: core/build.gradle.kts [hunk 1]
```diff
@@ -2,6 +2,33 @@ plugins {
...
+          licenses {
+            license {
+              name.set("The Apache License, Version 2.0")
+              url.set("http://www.apache.org/licenses/LICENSE-2.0.txt")
+            }
+          }
+        }
+      }
+    }
+  }
```
Comment (s-ayush2903) [inline core/build.gradle.kts:31]: Maybe we can extract it out as a common functionality(like [SpotlessConfig](https://github.com/google/android-fhir/blob/master/buildSrc/src/main/kotlin/SpotlessConfig.kt#L21), and rename file to `BuildConfig.kt` or something more semantic), as the config is nearly same to that of dc; 
https://github.com/google/android-fhir/blob/b337529fcbac32556c580bdd79b4aca43b9feaeb/datacapture/build.gradle.kts#L18-L37

Thoughts?

Comment (s-ayush2903) [reply core/build.gradle.kts:31]: Ah, I referred to extracting the publishing method in  `afterEvaluate{}`, not everything

Comment (jingtang10) [reply core/build.gradle.kts:31]: We can. Not a high priority though.

**Review by jingtang10 — APPROVED** [review]
Thanks @Andati! Can you please rename the PR title to "Use Maven Publish Plugin for the core module" to be more exact?

**Pull request merged**